todo_app/tests/test_end_to_end.py

```diff
@@ -0,0 +1,101 @@
+import os
+import pytest
+from dotenv import find_dotenv , load_dotenv
+from selenium import webdriver
+from selenium import webdriver
+from seleniumrequests import Chrome
+from selenium.webdriver.support.ui import Select
+from todo_app.data.api import TrelloAPI
+from threading import Thread
+from todo_app import app
+
+
+def create_test_board():
+    api = TrelloAPI()
+    name = "Test Board"
+    test_board = api.create_board(name)
+    return test_board.get('id')
+
+def create_test_list(board_id):
+    api = TrelloAPI()
+    name = "To Do"
+    test_list = api.create_list(name, board_id)
+    return test_list.get('id')
+
+def delete_temp_board(board_id):
+    api = TrelloAPI()
+    board_id = board_id 
+    api.delete_board(board_id)
+    
+@pytest.fixture(scope='module')
+def test_app():
+    # Load the env file
+    file_path = find_dotenv('.env')
+    load_dotenv(file_path, override=True)
+    
+    # Create the new board & update the board id environment variable
+    board_id = create_test_board()
+    list_id = create_test_list(board_id)
+    
+    os.environ['BOARD_ID'] = board_id    
+    os.environ['TO_DO_ID'] = list_id    
+     
+    # construct the new application
+    application = app.create_app()
+
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield app
+    
+    print("tear down")
+    # Tear Down
+    thread.join(1)
+    delete_temp_board(board_id)
+    
+@pytest.fixture(scope="module")
+def driver():
+    with webdriver.Chrome("/usr/local/bin/chromedriver") as driver:
```

Comment:
We shouldn't hardcode this path, otherwise it's tightly coupled to how a developer's machine is set up. If we leave that argument empty it will rely on the chromedriver being set somewhere on their PATH, which should be enough

Reply:
Done